### PR TITLE
updating the undo tree when using bracket operators slightly

### PR DESCRIPTION
**Pull request by xconverge**
Yay! We love PRs! 🎊

Please include a description of your change & check your PR against this list, thanks:

- [X] Commit message has a short title & issue references
- [X] Each commit does a logical chunk of work.  
- [x] It builds and tests pass (e.g `gulp tslint`)

More info can be found by clicking the "guidelines for contributing" link above.

### CODE REVIEW

File: src/mode/modeHandler.ts [hunk 1]
```diff
@@ -742,12 +742,31 @@ export class ModeHandler implements vscode.Disposable {
...
+    ranAction = ranAction && vimState.currentMode === ModeName.Normal;
+
+    // }])> keys all start a new undo state when directly next to an {[(< opening character
     const key = vimState.recordedState.actionKeys[vimState.recordedState.actionKeys.length - 1];
 
-    ranRepeatableAction = (ranRepeatableAction && vimState.currentMode === ModeName.Normal) ||
-                          (vimState.currentMode === ModeName.Insert && "<[({".indexOf(key) !== -1);
-    ranAction           = ranAction           && vimState.currentMode === ModeName.Normal;
+    if (vimState.currentMode === ModeName.Insert) {
+      const letterToTheLeft = TextEditor.getLineAt(vimState.cursorPosition).text[vimState.cursorPosition.character - 2];
```
Comment (johnfn) [inline src/mode/modeHandler.ts]: How does this work when `vimState.cursorPosition.character - 2` is less than 0?

File: src/mode/modeHandler.ts [hunk 2]
```diff
@@ -742,12 +742,31 @@ export class ModeHandler implements vscode.Disposable {
...
+      switch (key) {
```
Comment (johnfn) [inline src/mode/modeHandler.ts]: This is getting a little unwieldy, could you pull it into a separate function?

**Comment by xconverge** [comment]
Fixing the overrun still, had to get off the train :)

**Comment by xconverge** [comment]
Ok, should be a bit better now

**Comment by johnfn** [comment]
Thanks, @xconverge! :)

**Pull request merged**